hydra/_internal/core_plugins/importlib_resources_config_source.py

```diff
@@ -29,14 +30,16 @@ def __init__(self, provider: str, path: str) -> None:
     def scheme() -> str:
         return "pkg"
 
-    def load_config(self, config_path: str) -> ConfigResult:
-        normalized_config_path = self._normalize_file_name(config_path)
-        res = resources.files(self.path).joinpath(normalized_config_path)  # type:ignore
-        if not res.exists():
-            raise ConfigLoadError(f"Config not found : {normalized_config_path}")
-
-        with res.open(encoding="utf-8") as f:
-            header_text = f.read(512)
+    def _read_config(self, res: Any) -> ConfigResult:
+        try:
+            if sys.version_info[0:2] >= (3, 8) and isinstance(res, zipfile.Path):
+                f = res.open()
+                header_text = f.read(512)
+                if isinstance(header_text, bytes):
+                    header_text = header_text.decode("utf-8")
+            else:
+                f = res.open(encoding="utf-8")
+                header_text = f.read(512)
```

Comment:
Does this path also work for regular files?
If so, we should be able to just keep this and remove the else.

Reply:
Can you add a comment explaining why we need the else so we both remember?

Reply:
good idea! added a comment.